core/src/main/java/org/apache/accumulo/core/client/rfile/RFile.java

```diff
@@ -408,7 +408,9 @@ public static SummaryInputArguments summaries() {
      *          Configuration for summarizer to run.
      * @since 2.0.0
      */
-    WriterOptions withSummarizers(SummarizerConfiguration... summarizerConf);
+    default WriterOptions withSummarizers(SummarizerConfiguration... summarizerConf) {
```

Comment:
This is an interesting constraint on fluent interfaces. Since this should only matter to implementing classes, and we should be the only ones providing an implementation for these inner-interfaces which exist for fluent APIs, I don't think it's necessary to provide a default method like this. However, it's probably not going to hurt, either. Perhaps it's good practice to just always create these interfaces as default methods rather than abstract ones?

Reply:
Yeah I wasn't sure if we should but wanted to put this change out there for discussion.  I was trying to think of a scenario where adding this to the fluent API would break existing code...

Reply:
The default method is definitely a clever solution that helps with strict adherence to Semver. So, it's probably worth doing for that reason alone. I don't think it would affect users unless they implemented the interfaces in their own code, which I can't imagine a sane reason for doing, but people are unpredictable and have done crazier things. :smiley_cat:

Reply:
Haha true that!  It would also just be a good habit for us to start doing for all new methods to existing interfaces.